Project Management Process.md

```diff
@@ -0,0 +1,287 @@
+
+#### Example
+
+***An example pipeline could have the following stages:***
+
+```
+  ╭─────────────────────────────────── Pipeline ──────────────────────────────────────╮
+  │────── 1. ────│──── 2. ───│─── 3. ──│────── 4. ─────│─── 5. ───│──── 6. ───│── 7. ─│
+  │ Stages                                                                            │
+
+                                                ╭────────────╮
+──> Future Work ──> Backlog ──> Ready ──> In Progress ──> Review ──> Release ──> Done ──>
+         │             │          │             │            │ 
+         ╰─────────────╯          │             │            │
+                                  ╰───────── Blocked ────────╯
+    Phases
+  │───────── Backlog ────────│──────────── Development ───────────│──── Complete ─────│
+  ╰───────────────────────────────────────────────────────────────────────────────────╯
+```
+
+1. Future Work - the goal is identified and being fleshed out. It is not ready to be carried out at this moment. 
+2. Backlog - the goal is well defined but not planned to be carried out at this moment.
+3. Ready - the goal is planned to be carried out, waiting to be worked on.
+4. In Progress - the goal is actively being worked on, this moment.
+5. Review - the development of the goal is done, but a review needs to happen before it can be included in a release.
+6. Release - the goal is waiting to be inckuded in a an official release.
+7. Done - the goal is completed. No further action is necessary.
+
+* Blocked - the goal cannot proceed as something is preventing it to move down the pipeline.
+
+### Weekly updates
+To review and discuss progress of each project and the organization as a whole, we hold weekly update meetings.
+
+TODO: describe our weekly process. don't call it a "Sprint" anymore(?)
+
+### Roadmap updates
+To review and discuss progress of each project and the organization as a whole, we hold quarterly roadmap update meetings.
+
+TODO
+
+What:
+- Event to update and plan the organization's / project's roadmaps.
+- Identify projects' / organizations' needs and main efforts.
+- Commit to efforts and goals.
+- Produce a high level overview of a roadmap.
+
+Who: Product Owners, Project Leads, Project Managers
+How often: Every 3 months
+Output: Organization Roadmap (responsible: PM)
+
+This could be also a constant process instead of doing all that work tied to quarterly event. If the work was constant, we could still have an update event to go through the updated/new roadmap(s).
+
+## Implementation Guidelines
+TODO
+
+- automation
+  - Generating milestone documentation: https://github.com/ipfs/pm/pull/131#discussion_r73778673
+- github
+- waffle
```

Comment:
> We can make it so that it knows how to cache things and only use the requests to manipulate the issues.

It also does quite a lot of this, it tries to cache the request in local storage so if you revisit it, it does not have to redo all the requests.